src/include/sof/audio/format.h

```diff
@@ -45,9 +45,10 @@
 	 (qx + qy - qz) : INT32_MIN) : INT32_MAX)
 
 /* Convert a float number to fractional Qnx.ny format. Note that there is no
- * check for nx+ny number of bits to fit the word length of int.
+ * check for nx+ny number of bits to fit the word length of int. The parameter
+ * qy must be 31 or less.
  */
-#define Q_CONVERT_FLOAT(f, qy)  ((int)((f) * (1 << qy) + 0.5)) /* f is float */
+#define Q_CONVERT_FLOAT(f, qy) ((int32_t)((f) * ((int64_t)1 << qy) + 0.5))
```

Comment:
@lgirdwood This code is executed in C pre-processor. I once checked that there was no code size overhead of using this macro vs. hard coded fixed point defines. These have the same code size.

```
#define PI_Q3_29  Q_CONVERT_FLOAT(3.1416, 29)
#define PI_Q3_29 1686633657

```

Reply:
ok, so shouldn't we be casting f or qy to const ? This would alert any users who were not converting a constant (i.e. using preprocessor)

Reply:
Yes, that's important, good catch. We do not want to inject any hidden float math with this.

Reply:
done

Reply:
Sorry, I thought a little more on this and casting const wont work in all the time. Best to use a type check macro like the ones defined here.
https://stackoverflow.com/questions/11449632/checking-types-of-macro-parameters-at-compile-time
This would catch all errors in usage.